GOVERNANCE.md

```diff
@@ -1,36 +1,87 @@
 # Governance
 
-`haskell-cryptography` (herein referred to as the "Organisation") is a GitHub organisation that aims to provide a support structure to open-source maintainers producing cryptography libraries.
+`haskell-cryptography` (herein referred to as the "Organisation") is a GitHub organisation that aims to provide a support structure to open-source maintainers producing cryptography projects.
 
-This document elaborates on administrative and operational aspects of the Organisation.
+The Organisation has the following goals:
+
+* Prevent situations where too few people are supporting too critical projects for the ecosystem.
+* Prevent having a very closed group of people having too much control over every aspect of a project.
+* To encourage and support part-time, niche contributors to the projects.
+* To ensure continuity and stability through documentation, testing and benchmarking.
+* To listen to the needs of the community and industrial users, and lowering the barrier to industrial adoption.
+* To ensure that current security practices are easy for others and maintained whithin the ecosystem.
+
+Examples of situations that we aim to remediate to:
+
+> A single maintainer doing all the “boring work” suddenly becomes unavailable for reasons outside of the project.
+
+> A large number of people requesting a fix or feature not being listened to for personal disagreements with the core maintenance team.
+
+> Somebody wanting to contribute a feature but not knowing where to begin, how to approach it, because it is not explained anywhere.
+
+> A pull-request lands which creates massive performance regression when compiled with `-O2`, affecting a large number of downstream users.
+
+> Losing adoption opportunities because the most popular options are too entrenched and are maintained by stubborn people.
+
+> A security flaw is discovered by researchers and we need to urgently upgrade a project, in a possibly breaking way.
+
+---
+
+This document elaborates on administrative and operational aspects of the Organisation, designed to serve and fullfill these goals as a priority.
 
 ## Administrative aspect
 
-The Organisation's administrative aspects are handled by the Core Team. While a healthy overlap is appreciated between the Core Team and the contributors,
-An employee of the Haskell Foundation has a position in the Core Team as to facilitate ecosystem-related matters and discussion with other institutional parties.
+The Organisation's administrative aspects are handled by the Core Team. The Core team are also contributors to projects.
+
+### Bylaws
+
+1. Core Team members are expected to do various kind of tasks (not niche things)
+2. Core Team members are expected to be available long-term, and it if it's not the case, they are to be substituted or replaced
```

Comment:
This might need to be made more concrete too. 
```suggestion
2. Core Team members are expected to be available long-term, and it if it's not the case, the other Core Team members may temporarily or permanently reassign their position.
```
I don't like this wording either, but it any least says what the mechanism is.